features/map.go

```diff
@@ -120,29 +122,8 @@ func haveMapType(mt ebpf.MapType) error {
 		fd.Close()
 	}
 
-	switch {
-	// For nested and storage map types we accept EBADF as indicator that these maps are supported
-	case errors.Is(err, unix.EBADF):
-		if isMapOfMaps(mt) || isStorageMap(mt) {
-			err = nil
-		}
-
-	// ENOTSUPP means the map type is at least known to the kernel.
-	case errors.Is(err, sys.ENOTSUPP):
-		if mt == ebpf.StructOpsMap {
-			err = nil
-		}
-
-	// EINVAL occurs when attempting to create a map with an unknown type.
-	// E2BIG occurs when MapCreateAttr contains non-zero bytes past the end
-	// of the struct known by the running kernel, meaning the kernel is too old
-	// to support the given map type.
-	case errors.Is(err, unix.EINVAL), errors.Is(err, unix.E2BIG):
-		err = ebpf.ErrNotSupported
-	}
-
+	err = convertProbeError(mt, err)
```

Comment:
I think reusing this logic for flag detection is incorrect: we craft some very specific map attributes to elicit specific Errno. The same error isn't necessarily caused when flags are present. I'd suggest you leave this part untouched.

Reply:
Noted, this reverted in the latest commit